spec/models/vacancy_spec.rb

```diff
@@ -406,4 +406,19 @@
       end
     end
   end
+
+  describe "#bigquery_view_count" do
+    subject { build_stubbed(:vacancy) }
+    let(:response) { [{ number_of_unique_vacancy_views: 42 }] }
+    let(:big_query) { double("Bigquery") }
+
+    before do
+      allow(Google::Cloud::Bigquery).to receive(:new).and_return(big_query)
+      allow(big_query).to receive(:query).with(/SELECT number_of_unique_vacancy_views.*WHERE id="#{StringAnonymiser.new(subject.id)}"/m).and_return(response)
```

Comment:
should this regex allow for multiple 'WHERE' - i mean where... and... actually im confused why .with() is regex at all?

Reply:
Basically it being a regex implies "I want the query method to be called with a string that matches this regex".

I've not added the `AND` in because that's mostly a query optimisation on our end and it would still work without, and I didn't want to tie the test too closely to the SQL - but maybe I'll change it to do a proper string match to be more precise.